DllImportGenerator/DllImportGenerator/Marshalling/CustomNativeTypeMarshaler.cs

```diff
@@ -0,0 +1,272 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+#nullable enable
+
+namespace Microsoft.Interop
+{
+    class CustomNativeTypeMarshaler : IMarshallingGenerator
```

Comment:
We have been going with a double `l` for marshaller/marshalling (file name and variable suffix too).